contracts/products/SolaceCoverProductV3.sol

```diff
@@ -0,0 +1,423 @@
+     */
+    function maxCover() public view override returns (uint256 cover) {
+        return IRiskManager(riskManager).maxCoverPerStrategy(address(this));
+    }
+
+    /**
+     * @notice Returns the active cover limit in **USD** to 18 decimal places. In other words, the total cover that has been sold at the current time.
+     * @return amount The active cover limit.
+     */
+    function activeCoverLimit() public view override returns (uint256 amount) {
+        return IRiskManager(riskManager).activeCoverLimitPerStrategy(address(this));
+    }
+
+    /**
+     * @notice Determine the available remaining capacity for new cover.
+     * @return capacity The amount of available remaining capacity for new cover.
+     */
+    function availableCoverCapacity() public view override returns (uint256 capacity) {
+        capacity = maxCover() - activeCoverLimit();
+    }
+
+    /**
+     * @notice Returns true if the policy is active, false if inactive
+     * @param _policyID The policy ID.
+     * @return status True if policy is active. False otherwise.
+     */
+    function policyStatus(uint256 _policyID) public view override returns (bool status) {
+        return coverLimitOf[_policyID] > 0 ? true : false;
+    }
+
+    /**
+     * @notice Calculate minimum required account balance for a given cover limit. Equals the maximum chargeable fee for one epoch.
+     * @param _coverLimit The maximum value to cover in **USD**.
+     */
+    function minRequiredAccountBalance(uint256 _coverLimit) public view override returns (uint256 mrab) {
+        mrab = (maxRateNum * chargeCycle * _coverLimit) / maxRateDenom;
+    }
+
+    /**
+     * @notice Calculates the minimum amount of Solace Cover Dollars required by this contract for the account to hold.
+     * @param _policyholder The account to query.
+     * @return amount The amount of SCD the account must hold.
+     */
+    function minScpRequired(address _policyholder) external view override returns (uint256 amount) {
+          return minRequiredAccountBalance(coverLimitOf[policyOf[_policyholder]]) + debtOf[_policyholder];
+    }
+
+    /**
+     * @notice Returns the Uniform Resource Identifier (URI) for `policyID`.
+     * @param policyID The policy ID.
+     */
+    function tokenURI(uint256 policyID) public view virtual override returns (string memory uri) {
+        require(_exists(policyID), "invalid policy");
+        return string(abi.encodePacked(baseURI, Strings.toString(policyID)));
+    }
+
+    /**
+     * @notice Calculates the policy cancellation fee.
+     * @param policyID The policy id.
+     * @return fee The cancellation fee.
+    */
+    function calculateCancelFee(uint256 policyID) public view override returns (uint256 fee) {
+        // calculate cancellation fee
+        if (latestChargedTime > 0) {
+            return (minRequiredAccountBalance(coverLimitOf[policyID]) * ((block.timestamp - latestChargedTime) / 86400)) / chargeCycle;
```

Comment:
Are we dividing by different time units?
`((block.timestamp - latestChargedTime) / 86400)` => Time since last charge in days
`chargeCycle` => Charge cycle in seconds

Reply:
has this been addressed? @ssozuer

Reply:
This statement is removed.